### DISCUSSION TITLE
Liquidsoap basic conf problem

### DISCUSSION THREAD

**Discussion by s0t0na666** [savonet/liquidsoap#1771] [Q&A]
Hi everyone, I'm having trouble setting up my liquidsoap with icecast. I'm looking to have autoplay from the server and replace it automatically with input.harbor when it's on, otherwise autoplay should be playing files. Logs don't work for some reason :( Icecast is working just fine. Here's my config file for liquidsoap:

```
#!/usr/bin/liquidsoap
# Don't create a pidfile
set("init.daemon.pidfile",false)
# Create Log File
set("log.file",true)
set("log.stdout",false)
set("log.level",3)
set("log.file.perms",755)
set("log.file.path","/opt/liquidsoap/logs/liq.log")
set("log.unix_timestamps",true)
# DJ or Metadata IP Address
set("harbor.bind_addr","0.0.0.0")
# Port / Pass for Live DJs
live = input.harbor(id="live",port=8000,password="123456", "live")
# Find /home/admin/music/zolod/ -type f -name "*.mp3" > /etc/liquidsoap/music.m3u
# Path to playlist file which contains a list of local mp3's (/home/user/mp3/song.mp3)
playlist = playlist("/home/admin/music/")
# Path to backup track if other streams fail
backup = single("/home/admin/music/123.ogg")
# Do not monitor for radio silence and also specify the expected play order
radio = fallback(track_sensitive=false,[live, playlist, backup])
# Output to an Icecast Server
output.icecast(%vorbis,
  mount="/stream",
  host="localhost", port=8000, password="123456",
  radio)
```
When started liquidsoap gives me this message: 
```
[ogg @ 0x55af82b7cc80] Cannot identify new stream
[ogg @ 0x55af82b7cc80] failed to create or replace stream
[ogg @ 0x55af82b7cc80] Estimating duration from bitrate, this may be inaccurate
[ogg @ 0x55af82bded40] Cannot identify new stream
[ogg @ 0x55af82bded40] failed to create or replace stream
[ogg @ 0x55af82bded40] Estimating duration from bitrate, this may be inaccurate
notification: Speex header too small
```
Thanks in advance!

**Comment by toots** [comment]
Hi @s0t0na666, welcome to the community!

Are you getting these logs when trying to connect to `input.harbor`? If so, I would suggest to remove `ffmpeg` and install just `ogg` and `vorbis` (provided your stream is `vorbis` audio. Otherwise, you can install `opus` and etc). This would allow us to check if your issue is coming from the `ffmpeg` decoder.

Otherwise, I would suggest to add a little more details:
- What actions are you doing?
- What logs do you see? Maybe add `set("log.level",4)` to your script and post a longer extract, including before the action you are doing.
- What outcome do you see? Is the final stream not connecting to icecast? Streaming blank or noise?

Let us know!

**Comment by s0t0na666** [comment]
Thank you so much for replying!!!

I believe I was missing mp3 encoder, only encoder installed when previous log was created was vorbis. I'm reinstalling everything with opam (initially I put it on root now I'm reinstalling on admin user). I'll try to pull off some basic configuration, for example only autodj playlist and security file. I'll post results here. I'll be more than happy if you could send me some config.liq examples. 


What actions are you doing? > I'm not clear what are you referring to. Please explain.

   What logs do you see? Maybe add set("log.level",4) to your script and post a longer extract, including before the action you are doing. > logs are not being populated, most likely due to permissions, I'll add "log.level",4 and will adjust the permissions. 

 What outcome do you see? Is the final stream not connecting to icecast? Streaming blank or noise? So far I wasn't able to start liquidsoap, after receiving the message above I was experimenting but everytime when I tried to execute with command ./liquidsoap -c /etc/radio.liq, upon sending the command nothing happens, no message, nothing, just got back to admin@vsp:~$.

Thank you for your assistance, I can't wait to have this sorted out.

**Comment by s0t0na666** [comment]
Liquidsoap is just ignoring sudo iquidsoap -c /etc/liquidsoap/radio.liq > result is absolutely **nothing** no error message, no process i ps. 

I'm on admin user and I've aplied chmod -R 755 & chown admin:admin on all related liquidsoap folders. I tried to assign liquidsoap folders to liquidsoap user but nothing changed. 

Logs are still not operational. 

I'm suspecting that liquidsoap folders are not connecting since I made some custom folders and I might've messed up. I would appreciate if you could give me some guidelines regarding the liquidsoap files and folders and their locations.
 
/etc/liquidsoap/
/usr/bin/liquidsoap
~/.opam/4.0.8.0/lib/liquidsoap/etc
~/opam/4.0.8.0/lib/liquidsoap/share
~/.opam.4.0.8.0/bin/liquidsoap
~/liquidsoap/share 
                    /etc
                   /log

It's a mess







Any ideas? :/

**Comment by toots** [comment]
The `-c` option does not work the same way as with `/bin/sh`. If you want to start a script manually, all you have to do is:
```sh
% liquidsoap /path/to/script.liq
```

**Comment by s0t0na666** [comment]
> 
> 
> The `-c` option does not work the same way as with `/bin/sh`. If you want to start a script manually, all you have to do is:
> 
> ```shell
> % liquidsoap /path/to/script.liq
> ```

Hey, thanks for the advise. As you can see from my previous posts I don't have much experience with Linux servers and apps. I presume there another way to start the liquidsoap script, I would appreciate so much if you could point me in the right direction. 

Also, I'm confused with folder structure when installing with OPAM. Everything ends up on ~/.opam/4.0.8.0/lib & bin, should those files be distributed to /usr/bin, /etc/ in order for liquidsoap to run properly or the files can be used from .opam folder directly?
